packages/@o3r/extractors/src/core/comparator/metadata-files.helper.ts

```diff
@@ -0,0 +1,63 @@
+import type { SupportedPackageManagers } from '@o3r/schematics';
+import { readdirSync, readFileSync } from 'node:fs';
+import { coerce, gt } from 'semver';
+
+/**
+ * Returns a file from an npm package.
+ * @param packageRef Npm compatible package descriptor (version or range)
+ * @param filePaths List of files paths to extract
+ * @param packageManager Name of the package manager to use
+ */
+export async function getFilesFromRegistry(packageRef: string, filePaths: string[], packageManager: SupportedPackageManagers): Promise<{[key: string]: string}> {
+  const npmFileExtractor = await import(packageManager === 'npm' ? './package-managers-extractors/npm-file-extractor.helper' : './package-managers-extractors/yarn2-file-extractor.helper');
+  return npmFileExtractor.getFilesNpm(packageRef, filePaths);
+}
+
+// TODO : there is probably already an utility for this?
+/**
+ * Read and parses a JSON file
+ * @param metadataPath Path of the file
+ */
+export function getLocalMetadataFile<T>(metadataPath: string): T {
+  const migrationData = readFileSync(metadataPath).toString();
+  return JSON.parse(migrationData) as T;
+}
+
+/**
+ * Given a filename containing a version, extracts this version according to the input regex.
+ * @param filename Filename containing a version
+ * @param pattern Regex with a capturing group containing the version
+ */
+export function getVersionFromFilename(filename: string, pattern: RegExp): string | undefined {
+  return pattern.exec(filename)?.[1];
+}
+
+/**
+ * Given a path to a folder and a name pattern, returns the content of the file with the latest version in it's name.
+ * @param migrationMetadataPath Path to the folder dcontining the metadata
+ * @param pattern Regex with a capturing group containing the version
+ */
+export function getLatestMigrationMetadataFile(migrationMetadataPath: string, pattern: RegExp): string | undefined {
+  return readdirSync(migrationMetadataPath).reduce((latestVersionFile: string | undefined, fileName: string) => {
+    const version = getVersionFromFilename(fileName, pattern);
+    if (version && latestVersionFile) {
+      const previousLatest = pattern.exec(latestVersionFile)![1];
+      return gt(version, previousLatest) ? version : previousLatest;
+    } else {
+      return latestVersionFile;
+    }
+  }, undefined);
+}
+
+/**
+ * Returns the range of package versions from which we will get the previous version according to granularity.
+ * @param version Current package version
+ * @param granularity 'major' or 'minor'
+ */
+export function getVersionRangeFromLatestVersion(version: string, granularity: 'major' | 'minor'): string {
+  const semver = coerce(version);
```

Comment:
Might be a little permissive, but not sure it's a big deal here (ex: semver.valid(semver.coerce("42.6.7.9.3-alpha")); // '42.6.7')